src/machine/usb.go

```diff
@@ -212,6 +213,12 @@ func handleStandardSetup(setup usb.Setup) bool {
 		if setup.WValueL == 1 { // DEVICEREMOTEWAKEUP
 			isRemoteWakeUpEnabled = false
 		} else if setup.WValueL == 0 { // ENDPOINTHALT
+			if setup.WIndex < usb.NumberOfEndpoints && usbStallHandler[setup.WIndex&0x7F] != nil {
```

Comment:
Can you compare with the number of endpoints without masking the direction out? E.g.

```go
if idx := setup.WIndex & 0x7f; idx < usb.NumberOfEndpoints && usbStallHandler[idx] != nil
```

here and below?